File: go/src/leetcode/lc0009_test.go
package leetcode

import "testing"

func Test_isPalindrome(t *testing.T) {
	test0009(121, true, t)
	test0009(-121, false, t)
	test0009(10, false, t)
}

func test0009(x int, expected bool, t *testing.T) {
	if expected != isPalindrome(x) {
		t.Error()
	}
}

File: py/leetcode/lc0009_test.py
import unittest
from leetcode import lc0009 as lc


class Test(unittest.TestCase):
    def setUp(self):
        self.solution = lc.Solution()

    def test(self):
        self._test(121, True)
        self._test(-121, False)
        self._test(10, False)

    def _test(self, x, expected):
        self.assertEqual(expected, self.solution.isPalindrome(x))
